dask_ml/model_selection/_search.py

```diff
@@ -286,16 +304,65 @@ def normalize_params(params):
 
+def _generate_fit_params_key_vals(fit_params, keys_filtered=None):
+    '''
+        _generate_fit_params_key_vals returns keys and values in (name,full_name) and values format expected in CVcache functions
 
-def _get_fit_params(cv, fit_params, n_splits):
-    if not fit_params:
-        return [(n, None) for n in range(n_splits)]
+        fit_params: dict[str,Any] fit params dictionary from input to fit() call
+        keys_filtered: list[str] keys to filter from fit_params from output
+
+        returns: ( (fit_parameter_name,fit_parameter_full_name), fit_param_value )
+    '''
     keys = []
     vals = []
     for name, (full_name, val) in fit_params.items():
+        if keys_filtered is not None and name not in keys_filtered:
+            continue
         vals.append(val)
         keys.append((name, full_name))
-    return [(n, (cv_extract_params, cv, keys, vals, n)) for n in range(n_splits)]
+    return keys, vals
+
+def _get_weights_source(fit_params):
+    '''
+        _get_weights_source returns the weights source string. sklearn pipelines subscript the parameter names such that a "samples_weight" parameter must be resolved
+        to "clf__samples_weight" for some classifier "clf" in an sklearn pipeline. Also support eval_sample_weight as a priority source.
+
+        fit_params: dict[str,Any] fit params dictionary from input to fit() call
+        returns: str of source of test folds weights in fit_params dictionary
+    '''
+    eval_weight_source = None
+    if fit_params is None:
+        return eval_weight_source
+
+    # sklearn subscripting adds __ scores to step variables
+    for candidate_param in fit_params:
+        if "eval_sample_weight" in candidate_param:
+            eval_weight_source = candidate_param
+            break
+        elif "sample_weight" in candidate_param:
+            eval_weight_source = candidate_param
+            break
+    return eval_weight_source
+
+
+def _get_n_folds_fit_params(cv, fit_params, n_splits, keys_filtered=None):
+    '''
+        _get_n_folds_fit_params gets index given by fold number and the fit_params for that folds train folds and test fold
+
+        cv: (str) cv dask task name string,
+        fit_params: dict[str,Any] fit params dictionary from input to fit() call
+        n_splits: int: fold index
+        keys_filtered: list[str] keys to filter from fit_params from output
+
+        returns:
+            list( (int, tuple(Any) )
+            list of tuples of fold number and set of dask task that extracts that folds fit parameters for train folds and test folds
+            These tasks will be evaluated by scheduler later
+    '''
+    if not fit_params:
+        return [(n, (None,None)) for n in range(n_splits)]
+    keys, vals = _generate_fit_params_key_vals(fit_params, keys_filtered=keys_filtered)
+    return [(n, ( (cv_extract_params, cv, keys, vals, n, True), (cv_extract_params, cv, keys, vals, n, False)) ) for n in range(n_splits)]
```

Comment:
```suggestion
return [
    (
        n,
        (
            (cv_extract_params, cv, keys, vals, n, True),
            (cv_extract_params, cv, keys, vals, n, False)
        )
    ) for n in range(n_splits)
]
```

There's a couple of really dense, single line list comprehensions that need to be split up like this. 
Especially with the dense tuples improving readability here is important.

Reply:
Reformatted tuple and added comment on the stuff in tuple